router/image/templates/nginx.conf

```diff
@@ -15,43 +15,47 @@ http {
 
     types_hash_max_size 2048;
-    server_names_hash_max_size {{ or (.deis_router_serverNameHashMaxSize) "512" }};
-    server_names_hash_bucket_size {{ or (.deis_router_serverNameHashBucketSize) "64" }};
+    server_names_hash_max_size {{ if exists "/deis/router/serverNameHashMaxSize" }}{{ getv "/deis/router/serverNameHashMaxSize" }}{{ else }}512{{ end }};
+    server_names_hash_bucket_size {{ if exists "/deis/router/serverNameHashBucketSize" }}{{ getv "/deis/router/serverNameHashBucketSize" }}{{ else }}64{{ end }};
 
     include /opt/nginx/conf/mime.types;
     default_type application/octet-stream;
-    {{ if .deis_router_gzip }}
-    gzip {{ .deis_router_gzip }};
-    gzip_comp_level {{ or .deis_router_gzipCompLevel "5" }};
-    gzip_disable {{ or .deis_router_gzipDisable "\"msie6\"" }};
-    gzip_http_version {{ or .deis_router_gzipHttpVersion "1.1" }};
-    gzip_min_length {{ or .deis_router_gzipMinLength "256" }};
-    gzip_types {{ or .deis_router_gzipTypes "application/atom+xml application/javascript application/json application/rss+xml application/vnd.ms-fontobject application/x-font-ttf application/x-web-app-manifest+json application/xhtml+xml application/xml font/opentype image/svg+xml image/x-icon text/css text/plain text/x-component" }};
-    gzip_proxied {{ or .deis_router_gzipProxied "any" }};
-    gzip_vary {{ or .deis_router_gzipVary "on" }};
+    {{ if exists "/deis/router/gzip" }}
+    gzip {{ getv "/deis/router/gzip" }};
+    gzip_comp_level "{{ if exists "/deis/router/gzipCompLevel" }}{{ getv "/deis/router/gzipCompLevel" }}{{ else }}5{{ end }}";
+    gzip_disable "{{ if exists "/deis/router/gzipDisable" }}{{ getv "/deis/router/gzipDisable" }}{{ else }}\"msie6\"{{ end }}";
+    gzip_http_version "{{ if exists "/deis/router/gzipHttpVersion" }}{{ getv "/deis/router/gzipHttpVersion" }}{{ else }}1.1{{ end }}";
+    gzip_min_length "{{ if exists "/deis/router/gzipMinLength" }}{{ getv "/deis/router/gzipMinLength" }}{{ else }}256{{ end }}";
+    gzip_types {{ if exists "/deis/router/gzipTypes" }}{{ getv "/deis/router/gzipTypes" }}{{ else }}application/atom+xml application/javascript application/json application/rss+xml application/vnd.ms-fontobject application/x-font-ttf application/x-web-app-manifest+json application/xhtml+xml application/xml font/opentype image/svg+xml image/x-icon text/css text/plain text/x-component{{ end }};
+    gzip_proxied "{{ if exists "/deis/router/gzipProxied" }}{{ getv "/deis/router/gzipProxied" }}{{ else }}any{{ end }}";
+    gzip_vary {{ if exists "/deis/router/gzipVary" }}{{ getv "/deis/router/gzipVary" }}{{ else }}on{{ end }};
     {{ end }}
 
-    {{ $useFirewall := or .deis_router_firewall_enabled "false" }}{{ if eq $useFirewall "true" }}# include naxsi rules
+    {{ $useFirewall := "false" }}
+    {{ if exists "/deis/router/firewall/enabled" }}
+    {{ $useFirewall := getv "/deis/router/firewall/enabled" }}
```

Comment:
And changing the `$useFirewall` thing I see that I need to change all the template and use the `if exists` in a lot of places
Like this:
```
{{ if exists "/deis/router/firewall/enabled" }}{{ $useFirewall := getv "/deis/router/firewall/enabled" }}{{ if eq $useFirewall "true" }}
location /RequestDenied {
    return {{ if exists "/deis/router/firewall/errorCode" }}{{ getv "/deis/router/firewall/errorCode" }}{{ else }}400{{ end }};
}{{ end }}{{ end}}
```

which is horrible to read and understand